From ed0e81a673fbe09fa18622ff437c36075d33984d Mon Sep 17 00:00:00 2001
From: Luis Medinas <lmedinas@gnome.org>
Date: Mon, 10 Nov 2008 00:07:00 +0000
Subject: Fix for bgo #560077 non-working buttons on Add Song/Album windows. Patch

2008-11-09  Luis Medinas  <lmedinas@gnome.org>

        * data/glade/AddWindow.glade:
        * src/AddWindow.cs:

        Fix for bgo #560077 non-working buttons on Add Song/Album
windows.
        Patch from Priit Laes <amd@store20.com>.



svn path=/trunk/; revision=1236
---
diff --git a/data/glade/AddWindow.glade b/data/glade/AddWindow.glade
index 0e72749..4b3ef0e 100644
--- a/data/glade/AddWindow.glade
+++ b/data/glade/AddWindow.glade
@@ -19,8 +19,6 @@
   <property name="gravity">GDK_GRAVITY_NORTH_WEST</property>
   <property name="focus_on_map">True</property>
   <property name="has_separator">False</property>
-  <signal name="response" handler="OnWindowResponse" last_modification_time="Tue, 04 Jan 2005 17:48:15 GMT"/>
-  <signal name="delete_event" handler="OnWindowDeleteEvent" last_modification_time="Tue, 04 Jan 2005 17:48:11 GMT"/>
 
   <child internal-child="vbox">
     <widget class="GtkVBox" id="dialog-vbox1">
diff --git a/src/AddWindow.cs b/src/AddWindow.cs
index 0b602f6..eba501c 100644
--- a/src/AddWindow.cs
+++ b/src/AddWindow.cs
@@ -23,7 +23,7 @@ using System.Collections;
 
 namespace Muine
 {
-	public abstract class AddWindow : Gtk.Window
+	public abstract class AddWindow : Gtk.Dialog
 	{
 #region Enums
 #region Enums.ResponseType
@@ -116,6 +116,9 @@ namespace Muine
 			this.entry_container.Add (this.entry);
 
 			// List
+			this.Response               += OnWindowResponse;
+			this.DeleteEvent            += OnWindowDeleteEvent;
+
 			this.list.RowActivated      += OnRowActivated;
 			this.list.Selection.Changed += OnSelectionChanged;
 			scrolledwindow.Add (this.list);
--
cgit v0.8.2
